### PR TITLE
[Core] Use global variable to eliminate force thread termination in plasma

### PR DESCRIPTION
**Pull request by suquark**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by ericl** [comment]
What are the main changes of this PR, can you add a summary?

**Comment by suquark** [comment]
@ericl This PR moves the plasma store runner into a global variable, so we can stop the plasma store by stopping the eventloop when receiving a signal. Previously, we use a local variable for the plasma store, this become a problem when dealing with signals, so the previous solution is forcefully kill the store (and this causes different behavior compared to the plasma upstream).

Using a global variable is not a problem, because (1) there is only one plasma store thread in Raylet (2) the thirdparty dlmalloc library cannot be contained in a local variable, so even we use a local variable for plasma store, it does not mean better isolation.

**Review by ericl — APPROVED** [review]
Looks good. Consider putting that comment in the PR description in the future.

File: src/ray/object_manager/plasma/store_runner.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  std::string socket_name_;
+  int64_t system_memory_;
+  bool hugepages_enabled_;
+  std::string plasma_directory_;
+  std::string external_store_endpoint_;
+  std::unique_ptr<EventLoop> loop_;
+  std::unique_ptr<PlasmaStore> store_;
+};
+
+extern std::unique_ptr<PlasmaStoreRunner> plasma_store_runner;
```
Comment (ericl) [inline src/ray/object_manager/plasma/store_runner.h:29]: Can we add a comment about this?

Comment (suquark) [reply src/ray/object_manager/plasma/store_runner.h:29]: sure

**Comment by suquark** [comment]
lint test is green for these comments. ready for merge

**Pull request merged**